src/ctapipe/instrument/trigger.py

```diff
@@ -59,10 +59,16 @@ class SoftwareTrigger(TelescopeComponent):
     def __init__(self, subarray, *args, **kwargs):
         super().__init__(subarray, *args, **kwargs)
 
-        self._ids_by_type = {
-            str(type): set(self.subarray.get_tel_ids_for_type(type))
-            for type in self.subarray.telescope_types
-        }
+        # we are grouping telescopes by the str repr of the type
+        # this is needed since e.g. in prod6, LST-1 is slightly different
+        # from LST-2 to LST-4, but we still want the trigger to work with all
+        # LSTs
+        self._ids_by_type = {}
+        for tel in self.subarray.telescope_types:
+            tel_str = str(tel)
```

Comment:
maybe just a config variable that accept a list of telescopes would be the most robust then, even if it's more work and has to be specified for each Prod.  That would fully decouple the reliance on knowing the telescope type and which ones are compatible with each other.  

E.g. 

```yaml
StereoTrigger:
      participating_tel_ids = [1,2,3,4]   
```

Reply:
I agree just specifying IDs that belong to sich a trigger group is most straight forward, but for the case you mentioned I think you'd just exclude it from allowed_telescopes in the Subarray selection, right?

Reply:
The only case I can think of is that during commissioning of a new telescope, when multiple telescopes are already on-site, we may take some data where that telescope is in mono-trigger-mode (i.e. not connected to the hardware trigger), but where we might still want to do stereo analysis or use the SWAT. In that case, you'd for example have tel_id [1,2,3] in the hardware trigger sim, but not 4, even though 4 is also an LST. Of course I think that could still be configured with TelescopeParameters though, using the 'id' method instead of 'type'